src/net/socket_state.rs

```
@@ -118,7 +107,7 @@ impl SocketState {
 
     /// This will return all dropped packets from this connection.
     pub fn dropped_packets(&mut self, addr: SocketAddr) -> Result<Vec<Packet>> {
-        let connection = self.create_connection_if_not_exists(&addr)?;
+        let connection = self.connections.insert_if_not_exists(&addr)?;
         let mut lock = connection
             .write()
             .map_err(|_| NetworkError::AddConnectionToManagerFailed)?;
```

Should we bubble up this error?

---

No.